Add formal unittests, and expose match_main method

freakboy3742 opened the pull request.

freakboy3742 committed:
Added binary and packaging artefacts to gitignore file.

freakboy3742 committed:
Added formal unit tests.

freakboy3742 committed:
Added match_main interface.

PCManticore reviewed (commented):
This looks awesome! I don't have anything to comment other than that a test fails for me locally `FAIL: test_unicode_surrogate_pair (tests.test_diff.DiffTests)`. Also maybe we should add a Travis file to get these tests running in CI as well?

freakboy3742 committed:
Added Travis build configuration.

freakboy3742 committed:
Tweaked setup.py to trigger Travis build.

freakboy3742 committed:
Updated travis config to use a different approach to installing.

freakboy3742 committed:
Removed binary artefacts.

freakboy3742 committed:
Corrected Py3 prototype declaration.

freakboy3742 committed:
Corrected unicode rendering in test.

freakboy3742 committed:
Update equality assertion.

freakboy3742 committed:
Added Py3.4 to the travis config.

freakboy3742 committed:
Corrected Wide->narrow string cast.

freakboy3742 committed:
Hand-craft a version of codecvt_utf8

freakboy3742 committed:
Added some unicode examples to match_main tests.

freakboy3742 committed:
Use Python's string convert function (backported for Py2/3.4)

freakboy3742 commented:
@PCManticore I've corrected the error you were seeing (turns out it was a quirk of Apple unicode rendering), and added a Travis config to confirm builds work as expected.

PCManticore reviewed (approved):
This looks great, @freakboy3742 ! Let's :ship: it.
